src/dmd/dtemplate.d

```diff
@@ -8288,46 +8270,110 @@ struct TemplateStats
     uint numInstantiations;     // number of instantiations of the template
     uint uniqueInstantiations;  // number of unique instantiations of the template
 
+    TemplateInstances* allInstances;
+
     /*******************************
      * Add this instance
      */
-    static void incInstance(const TemplateDeclaration td)
+    static void incInstance(const TemplateDeclaration td,
+                            const TemplateInstance ti)
     {
+        // message(ti.loc, "incInstance %p %p", td, ti);
         if (!global.params.vtemplates)
             return;
         if (!td)
             return;
+        assert(ti);
         if (auto ts = cast(const void*) td in stats)
+        {
+            log(*ts, cast() ti);
             ++ts.numInstantiations;
+        }
         else
+        {
             stats[cast(const void*) td] = TemplateStats(1, 0);
+            log(stats[cast(const void*) td], ti);
+        }
     }
 
     /*******************************
      * Add this unique instance
      */
-    static void incUnique(const TemplateDeclaration td)
+    static void incUnique(const TemplateDeclaration td,
+                          const TemplateInstance ti)
     {
+        // message(ti.loc, "incUnique %p %p", td, ti);
         if (!global.params.vtemplates)
             return;
         if (!td)
             return;
+        assert(ti);
         if (auto ts = cast(const void*) td in stats)
             ++ts.uniqueInstantiations;
         else
             stats[cast(const void*) td] = TemplateStats(0, 1);
     }
-}
 
+    static private void log(ref TemplateStats ts,
```

Comment:
This would be better suited as a local method inside of `incUnique`

Reply:
Done